sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -3795,3 +3798,132 @@ def hpat_pandas_series_pct_change_impl(self, periods=1, fill_method='pad', limit
+              python -m hpat.runtests sdc.tests.test_series.TestSeries.test_series_op3
+              python -m hpat.runtests sdc.tests.test_series.TestSeries.test_series_op4
+              python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_operator_add*
+
+    Parameters
+    ----------
+    series: :obj:`pandas.Series`
+        Input series
+    other: :obj:`pandas.Series` or :obj:`scalar`
+        Series or scalar value to be used as a second argument of binary operation
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+        The result of the operation
+    """
+
+    _func_name = 'Operator add().'
+
+    ty_checker = TypeChecker('Operator add().')
+    ty_checker.check(self, SeriesType)
+
+    if not isinstance(other, (SeriesType, types.Number)):
+        ty_checker.raise_exc(other, 'pandas.series or scalar', 'other')
+
+    series_indexes_alignable = False
+    none_or_numeric_indexes = False
+    if isinstance(other, SeriesType):
+        if (other.index == string_array_type and self.index == string_array_type):
+            series_indexes_alignable = True
+
+        if ((isinstance(self.index, types.NoneType) or check_index_is_numeric(self))
+             and (isinstance(other.index, types.NoneType) or check_index_is_numeric(other))):
+            series_indexes_alignable = True
+            none_or_numeric_indexes = True
+
+    if isinstance(other, SeriesType) and not series_indexes_alignable:
+        raise TypingError('{} Not implemented for series with not-alignable indexes. \
+        Given: self.index={}, other.index={}'.format(_func_name, self.index, other.index))
+
+    # specializations for numeric series - TODO: support arithmetic operation on StringArrays
+    if (isinstance(other, types.Number)):
+        def hpat_pandas_series_add_scalar_impl(self, other):
+            return pandas.Series(self._data + other, self._index)
+
+        return hpat_pandas_series_add_scalar_impl
+
+    elif (isinstance(other, SeriesType)):
+
+        # optimization for series with default indexes, that can be aligned differently
+        if (isinstance(self.index, types.NoneType) and isinstance(other.index, types.NoneType)):
+            def hpat_pandas_series_add_impl(self, other):
+
+                if (len(self._data) == len(other._data)):
+                    return pandas.Series(numpy.asarray(self._data + other._data, numpy.float64))
+                else:
+                    min_data_size = min(len(self._data), len(other._data))
+                    max_data_size = max(len(self._data), len(other._data))
+                    new_data = numpy.empty(max_data_size, dtype=numpy.float64)
+                    new_data[:min_data_size] = self._data[:min_data_size] + other._data[:min_data_size]
+                    new_data[min_data_size:] = numpy.repeat(numpy.nan, max_data_size - min_data_size)
+
+                    return pandas.Series(new_data, self._index)
```

Comment:
@kozlov-alexey In some cases resulting series should have the same name as original one. This cases, seems to be are:
* series + scalar (i.e. A + 1)
* series + series, but both series has the same name

This is very important in order to implement DataFrame filters. I'm ok to do it in separate PR, but this must be done. Also tests for named series should be added

Reply:
Well, after some more investigations, it's not important for filters, but still we should be consistent with pandas

Reply:
It looks like, at the moment we can't support second case (when two series have the same name) since it is runtime condition, but whether series has name or not we should know at compile time.

Reply:
@AlexanderKalistratov Thank you! I think we should address this in a separate PR. BTW, I just checked if there's any difference between '1 + A' and 'A + 1' and there's no, in both cases resulting series takes name from A. I'm going to dig in pandas sources to clarify this behavior.